src/App.tsx

```diff
@@ -15,47 +15,89 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortType {
+  LENGTH = 'length',
+  ALPHABET = 'alphabet',
+  RESET = 'reset',
+}
 export const App: React.FC = () => {
+  const [type, setType] = useState('');
+  const [isReversed, setIsReversed] = useState(false);
+  const { LENGTH, ALPHABET, RESET } = SortType;
+
+  const setList = (goods : string[]) => {
+    const visibleGoods = [...goods];
+
+    switch (type) {
+      case ALPHABET:
+        visibleGoods.sort((a, b) => a.localeCompare(b));
+        break;
+      case LENGTH:
+        visibleGoods.sort((a, b) => a.length - b.length);
+        break;
+      case RESET:
+        return [...goods];
+      default:
+        break;
+    }
+
+    if (isReversed) {
+      visibleGoods.reverse();
+    }
+
+    return visibleGoods;
+  };
+
+  const resetHendler = () => {
+    setIsReversed(false);
+    setType('');
+  };
+
+  const goods = setList(goodsFromServer);
```

Comment:
So, I meant you can write smth like this
```suggestion
  const goods = useMemo(() => {
    const visibleGoods = [goodsFromServer];
    ....
  }, [goodsFromServer]
```